app/utils/token-info.ts

```diff
@@ -73,44 +73,57 @@ export async function getTokenInfo(
     return token;
 }
 
+async function getFullLegacyTokenInfoUsingCdn(
+    address: PublicKey,
+    chainId: ChainId
+): Promise<FullLegacyTokenInfo | undefined> {
+    const tokenListResponse = await fetch('https://cdn.jsdelivr.net/gh/solana-labs/token-list@latest/src/tokens/solana.tokenlist.json');
+    if (tokenListResponse.status >= 400) {
+        reportError(new Error('Error fetching token list from CDN'));
+        return undefined;
+    }
+    const { tokens } = await tokenListResponse.json() as FullLegacyTokenInfoList;
+    const tokenInfo = tokens.find(t => t.address === address.toString() && t.chainId === chainId)
+    return tokenInfo;
+}
+
 /**
  * Get the full token info from a CDN with the legacy token list
  * The UTL SDK only returns the most common fields, we sometimes need eg extensions
  * @param address Public key of the token
  * @param cluster Cluster to fetch the token info for
  */
-export async function getFullLegacyTokenInfoUsingCdn(
+export async function getFullTokenInfo(
     address: PublicKey,
     cluster: Cluster,
     connectionString: string
 ): Promise<FullLegacyTokenInfo | undefined> {
     const chainId = getChainId(cluster);
     if (!chainId) return undefined;
 
-    // We use the solana-labs token-list from CDN
-    // We can't use the solflare one because it only includes the coinGeckoId extension
-    const tokenListResponse = await fetch('https://cdn.jsdelivr.net/gh/solana-labs/token-list@latest/src/tokens/solana.tokenlist.json');
-    if (tokenListResponse.status >= 400) {
-        reportError(new Error('Error fetching token list from CDN'));
-        return undefined;
+    const [legacyCdnTokenInfo, sdkTokenInfo] = await Promise.all([
+        getFullLegacyTokenInfoUsingCdn(address, chainId),
+        getTokenInfo(address, cluster, connectionString)
+    ]);
+
+    if (!sdkTokenInfo) {
+        return legacyCdnTokenInfo;
```

Comment:
You can actually get a little bit more perf out of this if you:

* Kick both off at the same time
* Only `await` the legacy one if the other one for sure fails.

```ts
// Both of these will be kicked off synchronously, because of the way promises work.
const legacyCdnTokenInfoPromise = getFullLegacyTokenInfoUsingCdn(address, chainId);
const sdkTokenInfo = await getTokenInfo(address, cluster, connectionString);

if (!sdkTokenInfo) {
  // Only ever await this if the sdkTokenInfo for sure turned up falsey.
  return await legacyCdnTokenInfo;
}
```

Reply:
Hmm we use the legacy data on the other path as well though:

```
    return {
        address: sdkTokenInfo.address,
        chainId,
        decimals: sdkTokenInfo.decimals ?? legacyCdnTokenInfo?.decimals ?? 0,
        extensions: legacyCdnTokenInfo?.extensions,
        logoURI: sdkTokenInfo.logoURI ?? legacyCdnTokenInfo?.logoURI,
        name: sdkTokenInfo.name,
        symbol: sdkTokenInfo.symbol,
        tags
    }
```

Thinking about it more we could remove the fallback for decimals/logo, if you unset them on-chain, maybe we should respect that instead of using a legacy value. But while we use these extensions I think we're going to end up awaiting the `legacyCdnTokenInfo` on both return paths.

Reply:
Oh, haha.

![giphy-1](https://github.com/solana-labs/explorer/assets/13243/5089444b-dc9c-441e-bd56-9d78d59bcb22)